greenplumpython/func.py

```diff
@@ -12,26 +12,43 @@
 
 class FunctionCall(Expr):
     def __init__(
-        self, func_name: str, db: Database, args: Iterable[Expr] = [], as_name: Optional[str] = None
+        self,
+        func_name: str,
+        args: Iterable[Expr] = [],
+        as_name: Optional[str] = None,
+        db: Optional[Database] = None,
     ) -> None:
-        super().__init__(as_name)
+        table: Optional[Table] = None
+        for arg in args:
+            if isinstance(arg, Expr) and arg.table is not None:
+                if table is None:
+                    table = arg.table
+                elif table.name != arg.table.name:
+                    raise Exception("Cannot pass arguments from more than one tables")
+        db = table.db if table is not None else db
+        super().__init__(as_name, table=table, db=db)
         self._func_name = func_name
         self._args = args
-        self._db = db
 
     def __str__(self) -> str:
         args_string = ",".join([str(arg) for arg in self._args]) if any(self._args) else ""
         return f"{self._func_name}({args_string})"
 
     def to_table(self) -> Table:
         as_string = f"AS {self._as_name}" if self._as_name is not None else ""
-        ret_table = Table(f"SELECT * FROM {str(self)} {as_string}", db=self._db)
-        return Table(f"SELECT * FROM {ret_table.name}", parents=[ret_table], db=self._db)
+        from_caluse = f"FROM {self.table.name}" if self.table is not None else ""
+        parents = [self.table] if self.table is not None else []
+        orig_func_table = Table(
+            f"SELECT {str(self)} {as_string} {from_caluse}", db=self._db, parents=parents
+        )
+        return Table(
+            f"SELECT * FROM {orig_func_table.name}", parents=[orig_func_table], db=self._db
```

Comment:
assuming a `orig_func_table` is can be `from_item` in AST type.
what if the user does insert on this table?

this comment not for this PR, I think we should fix this before GreenplumPython becomes complex.

Reply:
Thanks for your feedback! Will consider this.